### PR TITLE
feat: add TaxiForm metadata to Program model

### PR DESCRIPTION
**Pull request by kalebjdavenport**
## Description
This PR adds a TaxiForm class which describes the fields needed to add a TaxiForm lead capture form to a Program about page.

## Related Ticket
[WS-3515](https://2u-internal.atlassian.net/browse/WS-3515)

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -3300,6 +3288,39 @@ def __str__(self):
...
+        help_text=_('The grouping of the Taxi Form (by 2U) that would be rendered instead of the hubspot capture form'),
+        max_length=50,
+        blank=True,
+        default='',
+    )
+    title = models.CharField(max_length=255, default=None, null=True, blank=True)
+    subtitle = models.CharField(max_length=255, default=None, null=True, blank=True)
+    post_submit_url = models.URLField(null=True, blank=True)
+
+    taxiForm = models.OneToOneField(
```
Comment (bseverino) [inline course_discovery/apps/course_metadata/models.py]: This should be a field on the `Program` model instead (also rename to `taxi_form` for consistency)

File: course_discovery/apps/course_metadata/models.py [hunk 2]
```diff
@@ -3300,6 +3288,39 @@ def __str__(self):
         return str(self.name) if self.name else str(self.uuid)
 
 
+class TaxiForm:
```
Comment (bseverino) [inline course_discovery/apps/course_metadata/models.py]: Remember to inherit a model class here (`TimeStampedModel` should be fine)

File: course_discovery/apps/course_metadata/tests/factories.py [hunk 3]
```diff
@@ -114,6 +114,17 @@ def facts(self, create, extracted, **kwargs):
             add_m2m_data(self.facts, extracted)
 
 
+class TaxiFormFactory(factory.django.DjangoModelFactory):
```
Comment (bseverino) [inline course_discovery/apps/course_metadata/tests/factories.py:118]: this looks good - you'll want to use this in the tests for program serializer https://github.com/openedx/course-discovery/blob/master/course_discovery/apps/api/tests/test_serializers.py

**Review by bseverino — CHANGES_REQUESTED** [review]
Couple things along with the inline comments:

1. Add an admin page for `TaxiForm`, and also add the new field for `Program` https://github.com/openedx/course-discovery/blob/master/course_discovery/apps/course_metadata/admin.py
2. Add the new field to `ProgramSerializer` https://github.com/openedx/course-discovery/blob/8d5e095953e620d5783a254a43238f0071c68096/course_discovery/apps/api/serializers.py#L1761
3. Run migrations locally to generate a migration file

You can reference older PRs (https://github.com/openedx/course-discovery/pull/3427/files is a good one)

**Review by Ali-D-Akbar — COMMENTED** [review]
~Changes look good to me~. Can you add model tests for `TaxiForm`?

**Review by Ali-D-Akbar — CHANGES_REQUESTED** [review]
There is no migration created for `TaxiForm`. Please create one by running the following inside discovery-shell:
`./manage.py makemigrations course_metadata`

File: course_discovery/apps/course_metadata/tests/factories.py [hunk 4]
```diff
@@ -115,6 +115,17 @@ def facts(self, create, extracted, **kwargs):
...
+    class Meta:
+        model = TaxiForm
+
+    form_id = FuzzyText()
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/course_metadata/tests/factories.py]: if this is an int, you can use `FuzzyInt` instead.

**Review by Ali-D-Akbar — APPROVED** [review]

**Review by ansabgillani — APPROVED** [review]

**Review by bseverino — APPROVED** [review]

**Pull request merged**